### PR TITLE
feat: refactor to clap derive, add fields for simulation flags

### PR DESCRIPTION
**Pull request by varex83**

### CODE REVIEW

File: src/cli.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+use clap::{Parser, Subcommand};
+
+#[derive(Debug, Parser)]
+#[command(name = "blockifier-tester")]
+pub struct Cli {
```
Comment (rodrigo-pino) [inline src/cli.rs:5]: Please update the README accordingly

File: src/main.rs [hunk 2]
```diff
@@ -1,26 +1,38 @@
...
+const RERUNNING_AFTER_FIXES: bool = true;
+
+fn should_run_block(block_number: &u64) -> bool {
+    RERUNNING_AFTER_FIXES
+        || (!Path::new(&format!("./results/{block_number}.json")).exists()
+            && !Path::new(&format!("./results/trace-{block_number}.json")).exists()
+            && ![
+                610508_u64, 610541_u64, 612572_u64, 612787_u64, 613138_u64, 613978_u64,
+            ]
+            .contains(block_number))
```
Comment (rodrigo-pino) [inline src/main.rs:33]: I think this is old code right? Why are you adding this back in? Why there is a fix set of blocks in the code?
What happened with the cache mechanism written by Pearson?

Also, you are missing the flag for the cache mechanism as well I think

File: src/main.rs [hunk 3]
```diff
@@ -114,20 +119,15 @@ async fn execute_traces(start_block: u64, end_block: u64, should_run_known: bool
...
-            }
+                .await
+                .unwrap();
+            // Note that this doesn't compare the reasons for failure or the result on a success
+            let successes = result.iter().filter(|result| result.is_correct()).count();
+            info!(
+                "Completed block {block_number} with {successes}/{} successes",
+                result.len()
+            );
+            log_block_report(block_number, result);
```
Comment (rodrigo-pino) [inline src/main.rs:130]: Why are you unsafely unwrapping `result`?

File: src/main.rs (hunk 2 above)
Comment (varex83) [reply src/main.rs:33]: It seems like some conflicts with local versioning were around, I will fix that 👍

File: src/main.rs (hunk 3 above)
Comment (varex83) [reply src/main.rs:130]: the same is here

**Review by rodrigo-pino — COMMENTED** [review]
Great, remember to update the readme. I believe the application name is now different?

**Comment by rodrigo-pino** [comment]
> Great, remember to update the readme. I believe the application name is now different?

Also, could you add to the README the flag of --always-run

**Review by rodrigo-pino — APPROVED** [review]

**Pull request merged**